### PR TITLE
Fixed marking cells with bombs for example `mines`

### PR DESCRIPTION
**Pull request by xilec**
Actually I don't know goal of this example and would this changes acceptable.

For fixing `auto_reveal` behaviour, it needs information about marked cells in model.
When `Marked` state moved to border's model. All data about cells state become in the model cells, therefore much easy was remove synchronization cell's state between view and model and cell's state from view and just share cells state from model to view (and draw it).

This PR depends on  #790

### CODE REVIEW

**Review by gyscos — COMMENTED** [review]
Thanks for the work, looks great! I admit I kinda forgot about this example, but it deserves more attention!

Just a couple comments, and it should be good to merge!

File: cursive/examples/mines/board/view.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        for (i, cell) in self.board.iter().enumerate() {
+            let x = (i % self.board.size.x) * 2;
+            let y = i / self.board.size.x;
+
+            let text = match (cell.state, cell.content, self.board.is_ended) {
+                (Closed, _, false) => " □",
+                (Marked, _, false) => " ■",
+                (Opened, Free(n), false) | (_, Free(n), true) => ["  ", " 1", " 2", " 3", " 4", " 5", " 6", " 7", " 8"][n],
+                (Opened, Bomb, false) | (_, Bomb, true) => "\u{01F4A3} "
```
[the remark this answers is not in the document]
Comment (xilec) [reply cursive/examples/mines/board/view.rs]: I agree, I removed redundant space

**Comment by xilec** [comment]
@gyscos, PR is ready to merge

**Review by gyscos — APPROVED** [review]
Thanks again for the work!

**Pull request merged**